packages/workflow/src/records/notification.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+import fetch from 'node-fetch'
+import { getEventType } from '@workflow/features/registration/utils'
+import { EVENT_TYPE, ValidRecord } from '@opencrvs/commons/types'
+import { NOTIFICATION_SERVICE_URL } from '@workflow/constants'
+import {
+  getInformantSMSNotification,
+  InformantNotificationName
+} from '@workflow/features/registration/smsNotificationUtils'
+
+type NotificationEvent = 'in-progress' | 'ready-for-review'
+
+export async function sendNotification(
+  action: NotificationEvent,
+  bundle: ValidRecord,
+  authToken: string
+) {
+  const eventType = getEventType(bundle).toLowerCase()
+  const res = await fetch(
+    new URL(`/${eventType}/${action}`, NOTIFICATION_SERVICE_URL).href,
+    {
+      method: 'POST',
+      body: JSON.stringify(bundle),
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${authToken}`
+      }
+    }
+  )
+  if (!res.ok) {
+    throw new Error(
+      `Forwarding bundle to notification service failed with [${
+        res.status
+      }] body: ${await res.text()}`
```

Comment:
This got me thinking if we should start introducing custom errors to which we could include a message, the original error and other metadata like the response text and status in this case? Something like `new HttpError(res, 'Forwarding...')`. What do you think?

Reply:
hm that should enable us to define what errors we are expecting and at the caller end handle those errors gracefully

Reply:
Btw why not just use the Boom errors?

Reply:
I can't seem to find anything better than internal server error (500) for this particular case?